main.go

```diff
@@ -228,7 +228,7 @@ func main() { //nolint:funlen
 		setupLog.Error(err, "unable to update from legacy operator version")
 	}
 
-	if err = upgrade.CleanupExistingResource(setupClient, platform); err != nil {
+	if err = upgrade.CleanupExistingResource(setupClient, platform, dscApplicationsNamespace, dscMonitoringNamespace); err != nil {
```

Comment:
I wonder if all the `upgrade.` parts could be executed as `manager.RunnableFunc`  like the default DSCI creation  (thanks @ykaliuta for doing it). This way we would (at least) be able to propagate the controller context down the call chain.

We can go even one step further and implement `CleanupRunnable` which will implement `manager.Runnable` and `inject.Client` interface and encapsulate this functionality. This way we will have direct access to the client from the owning struct too.

The same can be done for the legacy path.

Reply:
just saw your comment, if i understand this, you meant the part in the webhook
yes, i agree that is a good idea to pass client around than using as param.